### PR TITLE
[PS09] Miner serialization from token + authentication middleware

### PR DESCRIPTION
**Pull request by hanburger97**
Ticket | Title
---|---
[PS09](https://trello.com/c/eYnSUlXO/40-ps09-miner-serialization-from-token-authentication-middleware) | Miner serialization from token + authentication middleware

## Context
We are trying to handle the case where an incoming request can be from both our client (e.g. with an authorization token from Diamond in the header) or from an Strantum-compliant xmrrig or the like. We need to secure most miner RPC routes behind authentication and serialize the miner instance once authenticated.

## Description
* Implemented middleware for authentication on JSON-RPC routes
* Serialize a Miner using either cached JWT token or the Authorization JWT token in the Header

---

- [x] Added unit tests
- [x] Ready for merge
- [x] Safe to rollback

### CODE REVIEW

File: src/models/miner.model.js [hunk 1]
```diff
@@ -10,33 +10,28 @@ const DiamondApi = require('src/api/diamond.api.js');
...
      * Serialize from JSON or JWT token into a Miner instance
      * @param {object} data 
      */
-    async constructor(data) {
+    constructor(data) {
         
         this.address = data.address;
         this.name = data.name;
         this.id = data.accountId;
+        this.difficulty = config.get('pool:difficulty')
```
Comment (andyjung-dev) [inline src/models/miner.model.js]: Do we need to map difficulty to miner? Or do you mean the hashrate that the miner is providing

Comment (hanburger97) [reply src/models/miner.model.js]: That's a good suggestion, this is copied from the other pool, I guess the idea is that every miner has a start difficulty based on the port they connected to, and ca potentially update their difficulty based on their hashrate, 
🤷‍♂ IDK, what do you think, should be keep difficulty uniform for all miners on a specific port? @andyjung-dev

Comment (hanburger97) [reply src/models/miner.model.js]: Resolved, this is removed

**Review by andyjung-dev — APPROVED** [review]
LGTM

**Pull request merged**